### DISCUSSION TITLE
Explicitly closing a `Fieldset` (and freeing up underlying NetCDF files)

### DISCUSSION THREAD

**Discussion by VeckoTheGecko** [Parcels-code/Parcels#1213] [Q&A] [answered]
Is there a way of explicitly closing a `Fieldset`?

Everytime I rerun my code to create my `Fieldset` object in Jupyter I get a `PermissionError` and need to restart my kernel and run everything again:
`PermissionError: [Errno 13] Permission denied: <path_to_my_netcdf>`

I was looking in the [fieldset source](https://oceanparcels.org/gh-pages/html/_modules/parcels/fieldset.html) and couldn't find any method that looked to do the trick. Also calling `del fieldset` didn't free up the underlying NetCDF file.

Any advice/insight would be appreciated 🙂

**Comment by erikvansebille** [comment, marked as answer]
Hi @VeckoTheGecko, the opening and closing of Netcdf files is doen in [fieldfilebuffer.py](https://github.com/OceanParcels/parcels/blob/master/parcels/fieldfilebuffer.py), which does have an `exit` method: https://github.com/OceanParcels/parcels/blob/32600333a087492c177c10a11767ae45b11ea0b4/parcels/fieldfilebuffer.py#L56-L57 

Does that help?

**Reply by VeckoTheGecko** [reply]
Seems like that would do it. I can't test it as I can't recreate the error that I was having (I should have saved the state of my code when I encountered it).
